### PR TITLE
Address react-hooks/exhaustive-deps warnings

### PR DESCRIPTION
**Pull request by rafgarciaa**
[Enabling `eslint` to warn missing deps caused some components to throw warnings during compile time.](https://github.com/AdaptiveConsulting/ReactiveTraderCloud/pull/1212)

Screen Shot 2019-10-15 at 9 47 15 AM

** I purposely put `// eslint-disable-next-line react-hooks/exhaustive-deps` on some `useEffect` hooks since I assumed they are being used as a `componentDidMount()` (empty deps array).

### CODE REVIEW

File: src/client/src/apps/MainRoute/widgets/analytics/components/analyticsLineChart/AnalyticsLineChart.tsx [hunk 1]
```diff
@@ -80,7 +80,7 @@ const LineCharts: React.FC<LineChartProps> = React.memo(props => {
   useEffect(() => {
     const newOffset = offsetState + 1 === intervalWidth ? 0 : offsetState + 1
     setOffsetState(newOffset)
-  }, [props])
+  }, [props, offsetState])
```
Comment (rafgarciaa) [inline src/client/src/apps/MainRoute/widgets/analytics/components/analyticsLineChart/AnalyticsLineChart.tsx]: @fdore do we need `props` here as a dependency? It doesn't look right to me.

Comment (fdore) [reply src/client/src/apps/MainRoute/widgets/analytics/components/analyticsLineChart/AnalyticsLineChart.tsx]: `offsetState` should be enough

Comment (fdore) [reply src/client/src/apps/MainRoute/widgets/analytics/components/analyticsLineChart/AnalyticsLineChart.tsx]: actually, I don't think it makes sense to have a dep on`offsetState` here, given the only place were we change it is inside the effect

Comment (rafgarciaa) [reply src/client/src/apps/MainRoute/widgets/analytics/components/analyticsLineChart/AnalyticsLineChart.tsx]: should I leave it as an empty array then?

Comment (rafgarciaa) [reply src/client/src/apps/MainRoute/widgets/analytics/components/analyticsLineChart/AnalyticsLineChart.tsx]: Ah, got it. It's dependent on the changes in `model.seriesData`.

**Pull request merged**